File: app/src/main/java/com/example/dhu_timetable/ui/search/SearchActivity.java
package com.example.dhu_timetable.ui.search;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Spinner;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import com.example.dhu_timetable.R;
import com.example.dhu_timetable.ui.main.MainActivity;
import com.example.dhu_timetable.ui.subject.SubjectFragment;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.initialization.InitializationStatus;
import com.google.android.gms.ads.initialization.OnInitializationCompleteListener;

public class SearchActivity extends AppCompatActivity implements View.OnClickListener{

    private String email;
    private String subjectname;
    private String major;
    private String level;
    private String cyber;

    private EditText et_subjectname;

    private Button btn_confirm;
    private Button btn_reset;
    private Button btn_cancel;

    private Spinner spn_major;
    private Spinner spn_level;

    private CheckBox cb_cyber;
    private AdView mAdView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);

        MobileAds.initialize(this, new OnInitializationCompleteListener() {
            @Override
            public void onInitializationComplete(InitializationStatus initializationStatus) {
            }
        });

        mAdView = findViewById(R.id.search_adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);

        // 로그인 정보 유지를 위한 Intent
        Intent it = getIntent();
        email = it.getStringExtra("email");

        btn_confirm = findViewById(R.id.button_confirm);
        btn_reset = findViewById(R.id.button_reset);
        btn_cancel = findViewById(R.id.button_cancel);

        et_subjectname = findViewById(R.id.edit_subject);

        spn_major = findViewById(R.id.spinner_major);
        spn_level = findViewById(R.id.spinner_level);
        cb_cyber = findViewById(R.id.checkBox_cyber);

        btn_confirm.setOnClickListener(this);
        btn_reset.setOnClickListener(this);
        btn_cancel.setOnClickListener(this);

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
    }

    // 상단바 뒤로가기 버튼
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch(item.getItemId()){
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    // 검색화면 버튼 이벤트
    @Override
    public void onClick(View v) {
        Intent it;
        switch(v.getId()){
            case R.id.button_confirm : // 검색 데이터 변환 -> MainActivity로 전송
                subjectname = String.valueOf(et_subjectname.getText());
                if(subjectname.isEmpty())
                    subjectname = "%";
                else
                    subjectname = "%" + subjectname + "%";

                major = String.valueOf(spn_major.getSelectedItem());
                if(major.isEmpty())
                    major = "%";
                else
                    major = "%" + major + "%";

                level = String.valueOf(spn_level.getSelectedItem());
                if(level.length() > 1)
                    level = level.substring(0, 1);
                else if(level.isEmpty())
                    level = "%";

                if(cb_cyber.isChecked())
                    cyber = "Y";
                else
                    cyber = "%";

                Log.d("button_confirm : ", subjectname + major + level + cyber);

                // SubjectFragment로 데이터 바로 전송 --> 현재 미사용
                SubjectFragment.searchInstance("","",subjectname, level, major, cyber);

                it = new Intent();
                it.putExtra("subjectname", subjectname);
                it.putExtra("major", major);
                it.putExtra("level", level);
                it.putExtra("cyber", cyber);
                setResult(RESULT_OK, it);
                finish();
                break;

            case R.id.button_cancel:
                finish();
                break;

            case R.id.button_reset:
                et_subjectname.setText("");
                spn_major.setSelection(0);
                spn_level.setSelection(0);
                cb_cyber.setChecked(false);
                break;


        }
    }
}

File: app/src/main/java/com/dhu/dhu_timetable/ui/search/SearchActivity.kt
package com.dhu.dhu_timetable.ui.search

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import android.view.View
import android.widget.CheckBox
import android.widget.EditText
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import com.dhu.dhu_timetable.R
import com.dhu.dhu_timetable.databinding.ActivitySearchBinding
import com.dhu.dhu_timetable.ui.subject.SubjectFragment
import com.dhu.dhu_timetable.util.Conts.CYBER
import com.dhu.dhu_timetable.util.Conts.EMAIL
import com.dhu.dhu_timetable.util.Conts.LEVEL
import com.dhu.dhu_timetable.util.Conts.MAJOR
import com.dhu.dhu_timetable.util.Conts.SUBJECT_NAME
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.MobileAds

class SearchActivity : AppCompatActivity(), View.OnClickListener {
    private var email: String? = null
    private var subjectname: String? = null
    private var major: String? = null
    private var level: String? = null
    private var cyber: String? = null
    private var etSubjectName: EditText? = null
    private var spnMajor: Spinner? = null
    private var spnLevel: Spinner? = null
    private var cbCyber: CheckBox? = null
    private var mAdView: AdView? = null

    private lateinit var binding: ActivitySearchBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySearchBinding.inflate(layoutInflater)
        setContentView(binding.root)

        MobileAds.initialize(this) { }
        mAdView = binding.searchAdView
        val adRequest = AdRequest.Builder().build()
        mAdView?.loadAd(adRequest)

        // 로그인 정보 유지를 위한 Intent
        email = intent.getStringExtra(EMAIL)
        etSubjectName = binding.editSubject
        spnMajor = binding.spinnerMajor
        spnLevel = binding.spinnerLevel
        cbCyber = binding.checkBoxCyber

        binding.buttonConfirm.setOnClickListener(this)
        binding.buttonReset.setOnClickListener(this)
        binding.buttonCancel.setOnClickListener(this)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    // 상단바 뒤로가기 버튼
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    // 검색화면 버튼 이벤트
    override fun onClick(v: View) {
        when (v.id) {
            R.id.button_confirm -> {
                subjectname = etSubjectName?.text.toString()
                subjectname = subjectname.run {
                    when {
                        this.isNullOrEmpty() -> "%"
                        else -> "%$this%"
                    }
                }

                major = spnMajor?.selectedItem.toString()
                major = major.run {
                    when {
                        this.isNullOrEmpty() -> "%"
                        else -> "%$this%"
                    }
                }
                level = spnLevel?.selectedItem.toString()
                level = level.run {
                    when {
                        this.isNullOrEmpty() -> "%"
                        this.length > 1 -> "%${this.substring(0, 1)}%"
                        else -> "%"
                    }
                }

                cyber = if (cbCyber!!.isChecked) "Y" else "%"
                Log.d("button_confirm : ", subjectname + major + level + cyber)

                // SubjectFragment로 데이터 바로 전송 --> 현재 미사용
                SubjectFragment.searchInstance("", "", subjectname, level, major, cyber)
                Intent().apply {
                    putExtra(SUBJECT_NAME, subjectname)
                    putExtra(MAJOR, major)
                    putExtra(LEVEL, level)
                    putExtra(CYBER, cyber)
                    setResult(Activity.RESULT_OK, this)
                }
                finish()
            }
            R.id.button_cancel -> finish()
            R.id.button_reset -> {
                etSubjectName?.setText("")
                spnMajor?.setSelection(0)
                spnLevel?.setSelection(0)
                cbCyber?.isChecked = false

                Intent().apply {
                    putExtra(SUBJECT_NAME, "%")
                    putExtra(MAJOR, "%")
                    putExtra(LEVEL, "%")
                    putExtra(CYBER, "%")
                    setResult(Activity.RESULT_OK, this)
                }
                finish()
            }
        }
    }
}
